pageserver/src/walredo.rs

```diff
@@ -282,9 +288,12 @@ impl PostgresRedoManager {
         // next request will launch a new one.
         if result.is_err() {
             error!(
-                "error applying {} WAL records ({} bytes) to reconstruct page image at LSN {}",
+                "error applying {} WAL records {}..{} ({} bytes) to base image with LSN {} to reconstruct page image at LSN {}",
                 records.len(),
+				records.first().map(|p| p.0).unwrap_or(Lsn(0)),
+				records.last().map(|p| p.0).unwrap_or(Lsn(0)),
                 nbytes,
+				base_img_lsn,
                 lsn
```

Comment:
How about printing every WAL record, not just the first and last one?

Reply:
We know number of applied WAL records and LSN range. It will be possible then to check in WAL that all WAL records for this page were actually collected. I do not think that some extra information has to be logged here.

Reply:
I'm imagining that the bug with issue #2808 is that we are incorrectly skipping some delta layer files that contains WAL records we need. For example, there are four delta layers:

4000-5000
3000-4000
2000-3000
1000-2000

For some reason, our code skips the "2000-3000" layer, and only collects WAL records from the three other layers. Now the error message might say it tried to apply 10 records, for example, but there are 15 applicable records altogether in those four layers. How do you know whether it skipped "2000-3000" or "3000-4000".

Of course, it might be obvious from the actual case, or the bug might be something completely different, so this might not be important. But then again, it might be.

Reply:
My assumption is that it is still not possible to investigate this issue without WAL and layer map files.
One we have them it is quite easy to determine that some records are missed: I will just use `pg_waldump` utility and filter it's output, locating records referencing particular block. Them it will be not so difficult to determine which records are missed and if them belong to some particular layer - that it was for some reasons skiped during traversal.